broker/src/main/java/io/camunda/zeebe/broker/system/partitions/impl/steps/StreamProcessorPartitionStep.java

```diff
@@ -18,6 +20,12 @@
 
 public class StreamProcessorPartitionStep implements PartitionStep {
 
+  private final Role role;
+
+  public StreamProcessorPartitionStep(final Role role) {
```

Comment:
Couldn't we get these kind of informations from the current context ? I think the property is called currentRole?

Reply:
You are right. You can get the role from context. So the parameter is unnecessary. I will update it.